src/Gateways/AlipayHK/Support.php

```diff
@@ -32,34 +58,137 @@ public static function requestApi($data, $publicKey): Collection
         }
 
+        if (isset($dataToBeVerify['trade_status'])) {
+            return new Collection($data['response']['trade']);
+        }
+
         throw new GatewayException(
             'Get Alipay API error, please check raw data for further information.',
             $data,
             json_encode($data)
         );
     }
 
-    public static function verifySign($data, $publicKey = null, $sync = false, $sign = null): bool
+    public static function verifySign($data, $publicKey = null, $sync = false, $sign = null, $signType = "MD5"): bool
     {
-        $toBeVerified  = md5(
-            mb_convert_encoding(
-                urldecode(http_build_query($data)) . $publicKey,
-                'gb2312',
-                'utf-8'
-            )
-        );
+        $toBeVerified = "";
+        if ($signType == "MD5") {
+            $toBeVerified = md5(
+                mb_convert_encoding(
+                    urldecode(http_build_query($data)) . $publicKey,
+                    'gb2312',
+                    'utf-8'
+                )
+            );
+            return $sign === $toBeVerified;
+        }
+
+        if ($signType == "RSA") {//TODO Verify with alipay public key
+            return true;
+        }
+
+    }
+
+    /**
+     * Generate sign.
+     *
+     * @author yansongda <me@yansongda.cn>
+     *
+     * @return string
+     */
+    private static function generateRSASign($parmas, $privateKey = null, $separator): string
+    {
+        if (is_null($privateKey)) {
+            throw new InvalidConfigException('Missing Alipay Config -- [private_key]', 1);
+        }
+
+        if (Str::endsWith($privateKey, '.pem')) {
+            $privateKey = openssl_pkey_get_private($privateKey);
+        } else {
+            $privateKey = "-----BEGIN RSA PRIVATE KEY-----\n" .
+                wordwrap($privateKey, 64, "\n", true) .
+                "\n-----END RSA PRIVATE KEY-----";
+        }
+        openssl_sign(self::getSignContentForAppPayment($parmas, false, $separator), $sign, $privateKey, OPENSSL_ALGO_SHA1);
+
+        return base64_encode($sign);
+    }
+
+    /**
+     * Verfiy sign.
+     *
+     * @author yansongda <me@yansonga.cn>
+     *
+     * @param array $data
+     * @param string $publicKey
+     * @param bool $sync
+     * @param string|null $sign
+     *
+     * @return bool
+     */
+    public static function verifyRSASign($data, $publicKey = null, $sync = false, $sign = null): bool
+    {
+        if (is_null($publicKey)) {
+            throw new InvalidConfigException('Missing Alipay Config -- [ali_public_key]', 2);
+        }
+
+        if (Str::endsWith($publicKey, '.pem')) {
+            $publicKey = openssl_pkey_get_public($publicKey);
+        } else {
+            $publicKey = "-----BEGIN PUBLIC KEY-----\n" .
+                wordwrap($publicKey, 64, "\n", true) .
+                "\n-----END PUBLIC KEY-----";
+        }
+
+        $sign = $sign ?? $data['sign'];
```

Comment:
haha, you know the way to use this, but why [here](https://github.com/riseno/pay/pull/1/files#diff-4cda46bf6b7807d14a9c9a25892713adR80)